sympy/functions/elementary/exponential.py

```diff
@@ -456,6 +456,13 @@ def _eval_rewrite_as_tanh(self, arg):
         from sympy import tanh
         return (1 + tanh(arg/2))/(1 - tanh(arg/2))
 
+    def _eval_rewrite_as_sqrt(self, arg):
+        from sympy.functions.elementary.trigonometric import sin, cos
+        coeff, terms = arg.as_coeff_Mul()
```

Comment:
The code should be more like that in `eval`. Only a numeric `Mul` is handled, and then `coeff = arg.coeff(S.Pi*S.ImaginaryUnit)`.